### ISSUE TITLE
If a pool in the beginning of `withdrawQueue` is paused, depositor won't be able to redeem from other active pools also

### ISSUE THREAD

**Issue by NicolaMirchev** [shieldify-security/ion-protocol-c7bf309e9dba7e19eb81e456fa4cd3254e5e65b9#17]
## Severity
High

## Description
When a user wants to withdraw his stakings, he return his shares by calling `withdraw` or `redeem`, those two functions calls [_withdrawFromIonPool](https://github.com/shieldify-security/ion-protocol-c7bf309e9dba7e19eb81e456fa4cd3254e5e65b9/blob/e417da59cbc7c4e7b2b7643055d21154b29fe292/src/vault/Vault.sol#L453), which is then iterating over `withdrawQueue` from first added pool. The problem here is that if a pool is paused, the tx will revert [here](https://github.com/shieldify-security/ion-protocol-c7bf309e9dba7e19eb81e456fa4cd3254e5e65b9/blob/e417da59cbc7c4e7b2b7643055d21154b29fe292/src/vault/Vault.sol#L468) and user won't be able to receive his funds, even if there is available liquidity in next pools in the queue.

## Attack Scenario
Will write

## Recommendation
Check if the pool is paused and if it is, skip withdrawing from it

**Comment by junkim012** [comment]
This is true. A paused IonPool will lead to a temporary DoS of the vault's `withdraw` or `redeem`. In general, the `pause` on the IonPool is used for incident response scenarios such as a potential exploit or upon observing contract upgrades of integrated protocols, and that pausing vault interactions as a byproduct of the IonPool pause is intended behavior. 

However, we see that it would be more accurate to return zero values for depositable or withdrawable amounts when the pool is paused rather than to allow reverts. The likelihood is low, but the impact could be high, so we believe this issue is a medium. Will fix.
